test/e2e/throughput_test.go

```diff
@@ -0,0 +1,201 @@
+package e2e
+
+import (
+	"testing"
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/test"
+
+	pkgTest "github.com/knative/pkg/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"fmt"
+	//"time"
+	"net/http"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"time"
+	"github.com/knative/pkg/test/spoof"
+	"golang.org/x/sync/errgroup"
+	"sync/atomic"
+)
+
+func TestServiceThroughput(t *testing.T) {
+	var (
+		clients *test.Clients
+		logger  *logging.BaseLogger
+	)
+	// Create a service with concurrency 1 that could sleep for N ms.
+	// Limit its maxScale to 10 containers, wait for the service to scale down and hit it with concurrent requests.
+	logger = logging.GetContextLogger(t.Name())
+	clients = Setup(t)
+
+	helloWorldNames := test.ResourceNames{
+		Config: test.AppendRandomString(configName, logger),
+		Route:  test.AppendRandomString(routeName, logger),
+		Image:  "observed-concurrency",
+	}
+
+	configOptions := test.Options{
+		ContainerConcurrency: 1,
+	}
+
+	fopt := func(config *v1alpha1.Configuration) {
+		config.Spec.RevisionTemplate.Annotations = make(map[string]string)
+		config.Spec.RevisionTemplate.Annotations["autoscaling.knative.dev/maxScale"] = "10"
+	}
+
+	if _, err := test.CreateConfiguration(logger, clients, helloWorldNames, &configOptions, fopt); err != nil {
+		t.Fatalf("Failed to create Configuration: %v", err)
+	}
+
+	if _, err := test.CreateRoute(logger, clients, helloWorldNames); err != nil {
+		t.Fatalf("Failed to create Route: %v", err)
+	}
+
+	test.CleanupOnInterrupt(func() { TearDown(clients, helloWorldNames, logger) }, logger)
```

Comment:
you want this after create config, since if create route fails, you want to tear down the config.